app/data/filters.py

```diff
@@ -39,6 +47,26 @@ class DriverRecordFilter(RecordFilter):
     created_max = django_filters.IsoDateTimeFilter(name="created", lookup_expr='lte')
     created_by = django_filters.Filter(field_name='created_by', method='filter_created_by')
     weather = django_filters.Filter(field_name='weather')
+    outside_boundary = django_filters.Filter(field_name='geom', method='filter_outside_boundary')
+
+    def filter_outside_boundary(self, queryset, field_name, boundary_uuid):
+        """Filter records that fall outside the specified boundary."""
+        redis_conn = get_redis_connection('boundaries')
+        bounds_hexewkb = redis_conn.get(boundary_uuid)
+        one_month_seconds = 30 * 24 * 60 * 60
+        if bounds_hexewkb is None:
+            try:
+                boundary = Boundary.objects.get(pk=boundary_uuid)
+            except ValidationError:
+                raise ParseError('outside_boundary was passed an invalid UUID')
+            unioned_bounds = boundary.polygons.aggregate(all_polys=Union('geom'))['all_polys']
+            # Full resolution is very slow, so simplify down to roughly 100m (DRIVER is in lat/lon).
+            unioned_bounds = unioned_bounds.simplify(tolerance=0.001, preserve_topology=True)
```

Comment:
I do foresee this potentially being problematic, but I don't think we'll know whether or how it's problematic until we get some people using it. The World Bank described this feature as being necessary for finding points "in the middle of the ocean" so it sounds to me like they are more worried about obviously wrong points than just slightly-wrong points. That said, false positives could certainly clutter up the results and make finding the obviously-wrong points more difficult. Without actual users or data it was tough to know what the right balance was between speed and accuracy -- this level of simplification sped things up significantly and I didn't see any false positives when I compared to flagged points to the country boundary on the map (which is un-simplified), so I didn't figure it made much sense to spend a lot more time fiddling with it given the limited information we have at the moment.

Reply:
Yeah, that makes sense. I'm wondering if adding a buffer to the geometry before simplification might be something to consider to help tend towards having false negatives than false positives. Neither is great, but in my head I imagine this will be used in the record view most (Since in the map view the points in the middle of the ocean would be exactly as visible with this filter as without) so false positives would be a more significant problem than false negatives in terms of usability.

Reply:
That does seem logical; buffering can be an expensive operation though so I think if we do encounter problems with false negatives it might be better to lower the simplification tolerance rather than adding a different operation. I suspect that in some places, the quality of the shapefile being loaded in for the national boundary may be more of a limiting factor than the amount of simplification being applied.

Reply:
Yeah, definitely fine to wait to see how it works in use before going to far off the deep-end in terms of optimization 👍